pkg/asset/agent/manifests/clusterdeployment.go

```diff
@@ -39,21 +41,61 @@ func (*ClusterDeployment) Dependencies() []asset.Asset {
 }
 
 // Generate generates the ClusterDeployment manifest.
-func (i *ClusterDeployment) Generate(dependencies asset.Parents) error {
-	// TODO: generate from install-config
-	return i.finish()
+func (cd *ClusterDeployment) Generate(dependencies asset.Parents) error {
+	installConfig := &agent.OptionalInstallConfig{}
+	dependencies.Get(installConfig)
+
+	if installConfig.Config != nil {
+		clusterDeployment := &hivev1.ClusterDeployment{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ClusterDeployment",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      installConfig.Config.ObjectMeta.Name,
+				Namespace: installConfig.Config.ObjectMeta.Namespace,
+			},
+			Spec: hivev1.ClusterDeploymentSpec{
+				ClusterName: installConfig.Config.ObjectMeta.Name,
+				BaseDomain:  installConfig.Config.BaseDomain,
+				PullSecretRef: &corev1.LocalObjectReference{
+					Name: "pull-secret",
```

Comment:
This needs to have the same name as the Secret in the AgentPullSecret manifest.

I think the safest way to do this might be to depend on AgentPullSecret and call AgentPullSecret.ResourceName(). That way if the user can created a PullSecret already with a different name, we will just use it. I'm not sure if this kind of approach can be used everywhere without circular dependencies though.